File: src/lib/api/desktopSessions.ts
import { invoke } from "@tauri-apps/api/core";

/** Claude Desktop 某账号/组织下的会话分组。 */
export interface DesktopSessionAccount {
  /** 数据根类型：`default`（.../Claude）或 `managed`（.../Claude-3p）。 */
  rootKind: string;
  accountUuid: string;
  orgUuid: string;
  sessionCount: number;
  isCurrent: boolean;
}

/** 一次迁移的结果报告（与后端 `MigrateReport` 对应）。 */
export interface MigrateReport {
  fromAccount: string;
  fromOrg: string;
  toAccount: string;
  toOrg: string;
  /** 来源目录的会话数。 */
  sourceCount: number;
  /** 迁移前目标目录的会话数。 */
  destCountBefore: number;
  /** 来源中目标尚不存在的会话数（dryRun 时即「将新增」数量）。 */
  pending: number;
  /** 实际复制的会话数（dryRun 时为 0）。 */
  copied: number;
  /** 迁移后目标目录的会话数。 */
  destCountAfter: number;
  dryRun: boolean;
}

export interface MigrateOptions {
  fromRoot: string;
  fromAccount: string;
  fromOrg?: string | null;
  toRoot: string;
  toAccount: string;
  toOrg?: string | null;
  dryRun: boolean;
}

export const desktopSessionsApi = {
  async listAccounts(): Promise<DesktopSessionAccount[]> {
    return await invoke("list_desktop_session_accounts");
  },

  async migrate(options: MigrateOptions): Promise<MigrateReport> {
    const {
      fromRoot,
      fromAccount,
      fromOrg = null,
      toRoot,
      toAccount,
      toOrg = null,
      dryRun,
    } = options;
    return await invoke("migrate_desktop_sessions", {
      fromRoot,
      fromAccount,
      fromOrg,
      toRoot,
      toAccount,
      toOrg,
      dryRun,
    });
  },
};

File: src-tauri/src/commands/desktop_sessions.rs
#![allow(non_snake_case)]

use crate::desktop_sessions::{self, DesktopSessionAccount, MigrateReport};

/// 列出 Claude Desktop 各账号/组织下的会话分组（含当前登录标记）。
#[tauri::command]
pub async fn list_desktop_session_accounts() -> Result<Vec<DesktopSessionAccount>, String> {
    tauri::async_runtime::spawn_blocking(desktop_sessions::list_accounts)
        .await
        .map_err(|e| format!("扫描 Claude Desktop 会话失败: {e}"))?
        .map_err(|e| e.to_string())
}

/// 把来源账号的会话迁移（非破坏性复制）到目标账号；`toAccount` 缺省为当前登录账号。
/// `dryRun` 为真时只预览、不写入。
#[tauri::command]
pub async fn migrate_desktop_sessions(
    fromAccount: String,
    fromOrg: Option<String>,
    toAccount: Option<String>,
    toOrg: Option<String>,
    dryRun: bool,
) -> Result<MigrateReport, String> {
    tauri::async_runtime::spawn_blocking(move || {
        desktop_sessions::migrate(
            &fromAccount,
            fromOrg.as_deref(),
            toAccount.as_deref(),
            toOrg.as_deref(),
            dryRun,
        )
    })
    .await
    .map_err(|e| format!("迁移 Claude Desktop 会话失败: {e}"))?
    .map_err(|e| e.to_string())
}
